clang/test/CheckedC/inferred-bounds/widened-bounds-semantic-compare.c

```diff
@@ -0,0 +1,36 @@
+// Tests for bounds widening of _Nt_array_ptr's using function to semantically
```

Comment:
Should there also be some tests that involve constants?

Reply:
There are 3 reasons why I did not add tests involving constants yet:
1. Currently I only call CompareExprSemantic from BoundsAnalysis where it is invoked to compare only the bases of expressions. So we do not have any callers which invoke CompareExprSemantic for offsets (constants).
2. I intend to add tests for constants in my PR which does constant folding. That way I would be able to test constants with +/* operators.
3. A more full-fledged testing of constants can happen once I integrate the semantic comparison with SplitIntoBaseOffset logic.